### PR TITLE
Fix shading in jaxrs-client

### PR DESCRIPTION
**Pull request by uschi2000**
Don't merge yet; need to check that this works in our projects.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/375)

### CODE REVIEW

**Comment by uschi2000** [comment]
FYI @diogoholanda

**Comment by uschi2000** [comment]
Ready for review, @markelliot

File: jaxrs-clients/build.gradle [hunk 1]
```diff
@@ -50,8 +55,15 @@ shadowJar {
     relocate('zipkin', 'httpremoting.shaded.zipkin')
     relocate('org.joda.time', 'httpremoting.shaded.org.joda.time')
     relocate('okio', 'httpremoting.shaded.okio')
+    relocate('com.ea.agentloader', 'httpremoting.shaded.com.ea.agentloader')
```
Comment (markelliot) [inline jaxrs-clients/build.gradle]: Surprised we're relocating this.

File: jaxrs-clients/build.gradle [hunk 2]
```diff
@@ -50,8 +55,15 @@ shadowJar {
...
+    relocate('org.mortbay.jetty.alpn.agent', 'httpremoting.shaded.org.mortbay.jetty.alpn.agent')
```
Comment (markelliot) [inline jaxrs-clients/build.gradle]: And also this.

**Comment by uschi2000** [comment]
Other option is to add an external dependency on it. Fine with me either
way, but the current model (include in shadowjar with relocation) is broken.

**Comment by markelliot** [comment]
i think external dependency on that would be fine

**Comment by uschi2000** [comment]
Done. Turns out this actually had a bug (added dependency on http2-agent project and contained the relocated classes)

File: jaxrs-clients/build.gradle [hunk 3]
```diff
@@ -50,8 +55,13 @@ shadowJar {
     relocate('zipkin', 'httpremoting.shaded.zipkin')
     relocate('org.joda.time', 'httpremoting.shaded.org.joda.time')
     relocate('okio', 'httpremoting.shaded.okio')
+    relocate('org.jvnet.animal_sniffer', 'httpremoting.shaded.org.jvnet.animal_sniffer')
```
Comment (markelliot) [inline jaxrs-clients/build.gradle:58]: what pulls this in?

Comment (markelliot) [reply jaxrs-clients/build.gradle:58]: think this is probably related to the agentloader of the alpn thing, in whcih case also want to exclude it

**Comment by uschi2000** [comment]
Feign if I remember right. Good old animal sniffer.

**Comment by markelliot** [comment]
That's weird, because it's new here?

**Comment by uschi2000** [comment]
Double-checked, it comes from Feign 8.17.

**Comment by markelliot** [comment]
Cool, then +1

**Pull request merged**